pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -77,6 +77,20 @@ def __repr__(self) -> str:
     def name(self) -> str:
         return "Hadamard"
 
+    @property
+    def pauli_rep(self):
+        if self._pauli_rep is None:
+            self._pauli_rep = qml.pauli.PauliSentence(
+                {
+                    qml.pauli.PauliWord({self.wires[0]: "X"}): INV_SQRT2,
+                    qml.pauli.PauliWord({self.wires[0]: "Z"}): INV_SQRT2,
+                }
+            )
+        return self._pauli_rep
+
+    def __init__(self, wires: WiresLike, id: Optional[str] = None):
+        super().__init__(wires=wires, id=id)
+
```

Comment:
Why is this needed?

Reply:
I added it so that `self._pauli_rep` is explicitly set to `None` (within the super constructor) and because PauliX, PauliY, etc. already had the same call of a super constructor.

I will remove the constructor as there is also no difference in the test cases without it.

Reply:
Exactly, the super `__init__` is called anyways if there is no new one defined on a subclass :)